soroban-env-host/src/native_contract/token/contract.rs

```diff
@@ -71,6 +82,16 @@ pub trait TokenTrait {
         amount: i128,
     ) -> Result<(), HostError>;
 
+    fn burn(e: &Host, from: Signature, nonce: i128, amount: i128) -> Result<(), HostError>;
+
+    fn burn_from(
+        e: &Host,
+        spender: Signature,
+        nonce: i128,
+        from: Identifier,
+        amount: i128,
+    ) -> Result<(), HostError>;
+
     fn freeze(e: &Host, admin: Signature, nonce: i128, id: Identifier) -> Result<(), HostError>;
```

Comment:
Maybe go with `set_auth(bool authorized)`? More slim interface and still fits into 10 chars (with the limit lifted could be `set_authorized`.

Reply:
This is a pretty good idea. My only concern is that users could confused auth in the name with the sdk auth.

Reply:
That's actually my concern with the term 'authorized' in general. But if we stick to it, I don't see a big issue with using this name, given that everything else uses the same terminology anyway. FWIW in my current auth draft I use `authorize` function for authorizing the contract invocation, although maybe we should change it for something else.

Reply:
Updated to use set_auth.